test/lib/mutation_source_test.cc

```diff
@@ -323,9 +320,7 @@ static void test_slicing_and_fast_forwarding(tests::reader_concurrency_semaphore
 
     for (auto prange_size = 1u; prange_size < mutations.size(); prange_size += 2) {
         for (auto pstart = 0u; pstart + prange_size <= mutations.size(); pstart++) {
-            auto ms = boost::copy_range<std::vector<mutation>>(
-                mutations | boost::adaptors::sliced(pstart, pstart + prange_size)
-            );
+            auto ms = mutations | std::views::drop(pstart) | std::views::take(prange_size) | std::ranges::to<std::vector>();
```

Comment:
Can you elaborate on when the container's type can be deduced and when it cannot?
Here for example it is probably deduced, but in other places you write it explicitly.

Reply:
> Can you elaborate on when the container's type can be deduced and when it cannot? 

probably the best reference is the C++ standard in this regard, but i'd prefer not repeating it. probably what you are asking is why i am not consistent on putting the full type of `std::ranges::to<>` versus putting the template class's type.

because, I sometimes prefer simplicity over explicitness if the type of the result is obvious.

> Here for example it is probably deduced, but in other places you write it explicitly.

this is not accurate. because, well, i am not consistent in other places.

Reply:
Agree with @tchaikov. The goal is not to be consistent but to be clear. Sometimes clarity is achieved by writing down the full type. Sometimes it's clearer to omit the time as it avoids repetition.